server/storage/mvcc/kvstore_txn.go

```diff
@@ -114,15 +118,23 @@ func (tr *storeTxnRead) rangeKeys(ctx context.Context, key, end []byte, curRev i
 				zap.Int("len-values", len(vs)),
 			)
 		}
+		totalBytes += int64(len(vs[0]))
+		if ro.MaxBytes > 0 && totalBytes > ro.MaxBytes {
+			break
+		}
 		if err := kvs[i].Unmarshal(vs[0]); err != nil {
 			tr.s.lg.Fatal(
 				"failed to unmarshal mvccpb.KeyValue",
 				zap.Error(err),
 			)
 		}
+		count += 1
```

Comment:
`count == i` Please remove redundancy

Reply:
Not sure I understand this. We still need `count` outside of the loop to resize the slice at the end.

Reply:
Please see comment above about how similar for loop can be made simpler.